dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -46,63 +48,48 @@ const getSourcesForRoleAndResolution = (
 		: setsForRole.filter((set) => !set.src.includes('dpr=2'));
 };
 
-const getFallback = (
-	role: RoleType,
-	resolution: ResolutionType,
-	imageSources: ImageSource[],
-): string | undefined => {
-	// Get the sources for this role and resolution
-	const sources: SrcSetItem[] = getSourcesForRoleAndResolution(
-		imageSources,
-		role,
-		resolution,
-	);
+const getFallback = (sources: SrcSetItem[]): string | undefined => {
 	if (sources.length === 0) return undefined;
 	// The assumption here is readers on devices that do not support srcset are likely to be on poor
 	// network connections so we're going to fallback to a small image
 	return getClosestSetForWidth(300, sources).src;
 };
 
-const getSources = (
-	role: RoleType,
-	resolution: ResolutionType,
-	imageSources: ImageSource[],
-): string => {
-	// Get the sources for this role and resolution
-	const sources: SrcSetItem[] = getSourcesForRoleAndResolution(
-		imageSources,
-		role,
-		resolution,
-	);
+// The image sources we're provided use double-widths for HDPI images
+// We should therefor multiply the width based on if it's HDPI or not
+// e.g { url: '... ?width=500&dpr=2 ...', width: '1000' }
+export const getDesiredWidth = (width: number, hdpi: boolean) =>
+	hdpi ? width * 2 : width;
 
-	return sources.map((srcSet) => `${srcSet.src} ${srcSet.width}w`).join(',');
-};
+// Turn SrsSets into a string which can be passed to the source element
+export const getSourcesFromSrcSets = (sources: SrcSetItem[]) =>
+	sources.map((srcSet) => `${srcSet.src} ${srcSet.width}w`).join(',');
 
-/**
- *       mobile: 320
- *       mobileMedium: 375
- *       mobileLandscape: 480
- *       phablet: 660
- *       tablet: 740
- *       desktop: 980
- *       leftCol: 1140
- *       wide: 1300
- */
-
-const getSizes = (
+export const optimiseBreakpointSizes = (
```

Comment:
As discussed, it seems like this function is removing duplicates so perhaps the name can reflect that?